atm.rb

```diff
@@ -1,13 +1,64 @@
+# ATM by Eric Anderson for evaluation
 class Atm
-  STARTING_BALANCE = 100.0
-
-  def initialize(filename = "balance.txt")
+  def initialize(filename = 'balance.txt')
     @filename = filename
   end
 
   def run
-    balance = STARTING_BALANCE
+    balance = 100
+    if File.exist?(@filename)
+      @account = File.open("#{@filename}", 'r+')
+      balance = @account.to_a.last.to_f
+    end
+    menu(balance)
+  end
+
+  def menu(balance)
+    while !menu_display && (selection = gets.chomp.to_s.downcase) != 'q'
+      case selection
+      when 'd' then print_balance(balance = deposit(balance))
+      when 'w' then print_balance(balance = withdraw(balance))
+      when 'b' then print_balance(balance)
+      when 'h' then account_history(balance)
+      when 'q' then quit(balance)
```

Comment:
Is `when `q` ever being reached...?

As half of the condition says: `(selection = gets.chomp.to_s.downcase) != 'q'`